pysingfel/beam/sase.py

```diff
@@ -0,0 +1,59 @@
+import numpy as np
+from scipy import stats
+import matplotlib.pyplot as plt
+from matplotlib.colors import LogNorm
+
+from .base import Beam
+
+class SASEBeam(Beam):
+    def __init__(self, mu=None, sigma=None, n_spikes=0,
+                 *args, **kargs):
+        super(SASEBeam, self).__init__(**kargs)
+        self.mu = mu
+        self.sigma = sigma
+        self.n_spikes = n_spikes
+
+    def get_highest_wavenumber_beam(self):
+        """
+        For variable/polychromatic beam to return highest wavenumber.
+        """
+        return Beam(
+            wavenumber=self.wavenumber*1.5,
```

Comment:
explain in the code why multiply by 1.5

Reply:
1.5 is to ensure that the highest wavenumber falls within the energy range of the sase spectra.
e.g.  If the photon energy is set to 7120eV, the energy range is [7120-50, 7120+50] (the statistics is from the real sase).

Reply:
Since wavenumber is proportional to energy, the magic number 1.5 is reset to 1.01. (~7170/7120~1.007) in that Gaussian truncates after 4-5 sigmas.